nilearn/regions/region_extractor.py

```diff
@@ -0,0 +1,487 @@
+"""
+Better brain parcellations for Region of Interest analysis
+"""
+
+import nibabel
+import numpy as np
+
+from scipy.ndimage import label
+from scipy.stats import scoreatpercentile
+
+from sklearn.base import clone
+from sklearn.externals.joblib import Memory
+
+from .. import masking
+from ..input_data import NiftiMasker, NiftiMapsMasker
+from ..input_data.nifti_masker import filter_and_mask
+from .._utils import check_niimg, check_niimg_3d, check_niimg_4d
+from .._utils.extmath import fast_abs_percentile
+from .._utils.class_inspect import get_params
+from ..image import iter_img, new_img_like, resample_img
+from ..image.image import _smooth_array
+from .._utils.niimg_conversions import concat_niimgs, _check_same_fov
+from .._utils.compat import _basestring
+from ..externals.skimage import peak_local_max, random_walker
+
+
+def foreground_extraction(maps_img, mask_img=None, threshold='auto',
+                          thresholding_strategy='percentile'):
+    """ A function which keeps the most prominent regions of the maps,
+    denoted as foreground objects/regions extraction.
+
+    Parameters
+    ----------
+    maps_img: a 3D/4D Nifti like image/object
+        a path or filename or an image consists of statistical maps or atlas
+        maps.
+
+    mask_img: a Nifti like image/object, default is None
+        If given, mask image is applied directly on the input data 'maps_img'.
+        For example, if interested to consider only one particular brain region
+        then mask image to that particular brain region should be provided.
+        If none, masking will not be applied.
+
+    threshold: a float (t-value or ratio [0., 100.]), 'auto', default is 'auto'
+        a value used to threshold the maps to keep the most meaningful voxels.
+
+        If float representing as a t-value, a t-statistic value should be submitted
+        to keep the voxel intensities which are above than this value.
+        Mostly suitable, if user wants to keep the voxels based on raw statistic value.
+        In this case, user should know exactly which are the voxels to be kept by
+        visually checking the intensity value.
+
+        If float representing as a ratio, this value is used to keep the more intense
+        voxels across all maps. The most meaningful voxels are kept by multiplying
+        this value with the total size of the voxels.
+
+        If default 'auto', a pre-defined float value set to 0.8 will be used
+        in representing as a ratio.
+
+    thresholding_strategy: string {None, 'ratio_n_voxels', 'percentile'}, \
```

Comment:
This is redundant with the previous parameter. The threshold depends on the type of the previous parameter:
- float -> ratio_n_voxels
- string "XX%" -> percentile
- 'auto' -> automatic strategy
- None -> no thresholding

This is not only disturbing for the user but you will have to do complex checking to see if both parameters matches.